src/react/databrowser/objects/ObjectListTable.jsx

```diff
@@ -35,11 +36,20 @@ type Props = {
 export default function ObjectListTable({ objects, bucketName, toggled, isVersioningType, prefixWithSlash }: Props){
     const dispatch = useDispatch();
     const nextMarker = useSelector((state: AppState) => state.s3.listObjectsResults.nextMarker);
-    const isItemLoaded = index => {
-        const shouldRefetch = nextMarker && index === objects.size - 1;
+    const loading = useSelector((state: AppState) => state.networkActivity.counter > 0);
```

Comment:
in order to simplify a bit the application you can create a `useTypedSelector` like so : 
```js
export const useTypedSelector: <TSelected>(
  selector: (state: AppState) => TSelected,
  equalityFn?: (left: TSelected, right: TSelected) => boolean
) => TSelected = useSelector;
```

That you can then reuse everywhere without the necessity to type explicitly the state parameter

Reply:
Yes, I have a plan to do that using the helper type: `TypedUseSelectorHook`.